zngnative/coerce.go

```diff
@@ -11,48 +11,26 @@ import (
 // resulting coerced value is written to out, and true is returned. If
 // the value cannot be coerced, then false is returned.
 func CoerceToFloat64(in zng.Value) (float64, bool) {
-	var out float64
-	var err error
-	switch in.Type.ID() {
-	default:
+	native, err := ToNativeValue(in)
```

Comment:
Are the changes to this file separate cleanup?

Reply:
Not entirely, the CoerceNativeTo... functions are new, then there's some shuffling of code to implement the existing CoerceTo... functions in terms of their CoerceNativeTo... equivalents.  While I was there I made the float coercion more consistent with integer (eg removed the bool->float coercion which wasn't listed on the type coercion wiki page)